pkg/module/go_get_fetcher.go

```diff
@@ -19,11 +19,16 @@ type goGetFetcher struct {
 }
 
 // NewGoGetFetcher creates fetcher which uses go get tool to fetch modules
-func NewGoGetFetcher(goBinaryName string, fs afero.Fs) Fetcher {
+func NewGoGetFetcher(goBinaryName string, fs afero.Fs) (Fetcher, error) {
+	cmd := exec.Command("which", goBinaryName)
```

Comment:
same thing with PATH

Reply:
Earlier i had the `stat` implementation, the problem was, it would require the path of `vgo`, and we can't give simply `vgo`, even though its in PATH but `which vgo` will succeed. 

The tradeoff here is, If `gobin` is not in path, we've to give the absolute path.

Reply:
@devdinu  @michalpristas why not just do this: 
```golang
func NewGoGetFetcher(goBinaryName string, fs afero.Fs) (Fetcher, error) {
  err := exec.Command(goBinaryName).Run()
  if _, ok := err.(*exec.ExitError); !ok {
    return nil, err
  }
  ... // go binary exists
}

Reply:
and side note, we should add an errors.Op and wrap it :)

Reply:
ok i understood that this PR is about early failure considering that goBinaryName is correct either `vgo` or absolute path and verifying that it is existent. 
maybe we can do something like this
- which vgo - should succ for `vgo` fail for vgo not within PATH
- on ^failure: stat goBinPath, if exists, ok, if not err
?

Reply:
@marwan-at-work looks good. it will fail even if vgo exists but does not have +x which would pass with stat and which

Reply:
@michalpristas that's a good thing right?